contracts/Turnstone.vy

```diff
@@ -0,0 +1,108 @@
+# @version 0.3.3
+"""
+@title Turnstone
+@author Volume.Finance
+"""
+
+MAX_VALIDATORS: constant(uint256) = 128
+MAX_BATCH: constant(uint256) = 128
+POWER_THRESHOLD: constant(uint256) = 2_863_311_530 # 2/3 of 2^32
+TURNSTONE_ID: immutable(bytes32)
+
+struct ValsetArgs:
+    validators: DynArray[address, MAX_VALIDATORS]
+    powers: DynArray[uint256, MAX_VALIDATORS]
+    nonce: uint256
+
+struct LogicCallArgs:
+    logicContractAddress: address
+    payload: Bytes[1024]
+
+struct Signature:
+    v: uint256
+    r: uint256
+    s: uint256
+
+event UpdateCheckpoint:
+    new_checkpoint: bytes32
+
+last_checkpoint: public(bytes32)
+last_valset_nonce: public(uint256)
+
+@external
+def __init__(turnstone_id: bytes32, validators: DynArray[address, MAX_VALIDATORS], powers: DynArray[uint256, MAX_VALIDATORS]):
```

Comment:
```suggestion
def __init__(turnstone_id: bytes32, valset: ValsetArgs):
```

That was we have a single way of representing valsets.

Reply:
We don't need `valset_id` in the constructor. And it is for saving gas.

Reply:
But we should use valset_id in the constructor. valset_id is generated on the paloma chain